src/rule-unused-fields.js

```diff
@@ -129,28 +147,88 @@ function rule(context) {
     }
   }
 
+  function getEdgesAndNodesWhiteListFunctionCallArguments(calls) {
+    return calls.flatMap(call =>
+      call.arguments.map(arg => {
+        if (arg.type === 'Identifier') {
```

Comment:
this piece is hard to understand what it does, would be nice to add some doc comments

Reply:
I'll think about how to document it, it's kind of low level stuff for AST but basically this function tries to gather variable names for the function calls.